v0/destinations/engage/transform.js

```diff
@@ -0,0 +1,261 @@
+  return responseBuilder(
+    payload,
+    endpoint,
+    ConfigCategories.TRACK.method,
+    Config
+  );
+};
+const pageResponseBuilder = (message, Config) => {
+  const uid = getUID(message);
+  const endpoint = `${ConfigCategories.PAGE.endpoint.replace("uid", uid)}`;
+
+  const { properties } = message;
+  const payload = {};
+  const meta = refinePayload(properties, ConfigCategories.PAGE.genericFields);
+  const pagePayload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.PAGE.name]
+  );
+  const mergedPayload = { ...meta, ...pagePayload };
+  payload.properties = mergedPayload;
+  payload.timestamp = getFieldValueFromMessage(message, "timestamp");
+  const name = generatePageName(message);
+  payload.event = name;
+  return responseBuilder(
+    payload,
+    endpoint,
+    ConfigCategories.PAGE.method,
+    Config
+  );
+};
+
+// Engage Api Doc for group calls Ref: https://engage.so/docs/api/lists
+const groupResponseBuilder = (message, Config) => {
+  const { groupId } = message;
+  if (!groupId) {
+    throw new ErrorBuilder()
+      .setMessage("Group Id can not be empty.")
+      .setStatus(400)
+      .setStatTags({
+        destType: DESTINATION,
+        stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.SCOPE,
+        meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.META.BAD_PARAM
+      })
+      .build();
+  }
+  let uid = getDestinationExternalID(message, "engageId");
+  if (!uid) {
+    uid = getFieldValueFromMessage(message, "userIdOnly");
+  }
+  const traits = getFieldValueFromMessage(message, "traits");
+  if (
+    message.traits.operation &&
+    message.traits.operation !== "remove" &&
+    message.traits.operation !== "add"
+  ) {
+    throw new ErrorBuilder()
+      .setMessage(
+        `${message.traits.operation} is invalid for Operation field. Available are add or remove.`
+      )
+      .setStatus(400)
+      .setStatTags({
+        destType: DESTINATION,
+        stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.SCOPE,
+        meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.META.BAD_PARAM
+      })
+      .build();
+  }
+  const operation = traits.operation ? traits.operation : "add";
+  if (!uid && operation === "remove") {
+    throw new ErrorBuilder()
+      .setMessage("engageID is required for remove operation.")
+      .setStatus(400)
+      .setStatTags({
+        destType: DESTINATION,
+        stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.SCOPE,
+        meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.META.BAD_PARAM
+      })
+      .build();
+  }
+  let { method } = ConfigCategories.GROUP;
+  let endpoint = `${ConfigCategories.GROUP.endpoint.replace("id", groupId)}`;
+  const subscriberStatus = traits.subscriberStatus || true;
```

Comment:
Is this subscriber means the user who is going to subsribe to the group? If yes, then subscriberStatus should come in `context.traits`